jdd.js

```diff
@@ -1088,3 +1085,130 @@ jQuery(document).ready(function() {
         }
     });
 });
+
+// utilites
+// 
+/**
+ * Fixing typeof
+ * takes value and returns type of value
+ * @param  value 
+ * return typeof value
+ */
+function getType(value) {
+    if ((function () { return value && (value !== this); }).call(value)) {
+        //fallback on 'typeof' for truthy primitive values
+        return typeof value;
+    }
+    return ({}).toString.call(value).match(/\s([a-z|A-Z]+)/)[1].toLowerCase();
+}
+/**
+ * Iterate over array of objects and call given callback for each item in the array
+ * Optionally may take this as scope
+ * 
+ * @param array 
+ * @param callback 
+ * @param optional scope 
+ */
+function forEach(array, callback, scope) {
```

Comment:
Why do we need a custom `forEach` function and why are we defining it so far down in the file when it's used higher up?

Reply:
Array.prototype.forEach only support iterating over arrays however we need to iterate over nodeList which is an array like structure. Array.prototype.forEach doesn't support iterating over nodeList. So I've used a custom method. I could've used Array.prototype.forEach.call however I already have gone through this https://toddmotto.com/ditch-the-array-foreach-call-nodelist-hack/. So I have crated my own custom forEach function which enables iterating over array like objects.